gadopt/energy_solver.py

```diff
@@ -7,12 +7,21 @@
 from numbers import Number
 from typing import Any, Optional
 
-from firedrake import Constant, DirichletBC, Function, Jacobian, TensorFunctionSpace, dot
-
+from firedrake import (
```

Comment:
firedrake imports is not consistent across the different modules.... Probably not a big issue but probably better be consistent

Reply:
Consistency would be nice, yes. See other related comments.